### ISSUE TITLE
Can't locate LatexIndent/Document.pm

### ISSUE THREAD

**Issue by castn** [cmhughes/latexindent.pl#363]
Hi,
installed the new version 3.17.2 via conda from conda-forge. By running the script, I got the following message `Can't locate LatexIndent/Document.pm in @INC (you may need to install the LatexIndent::Document module)`. I am not quite sure what that means. It looks like it has something to do with `use LatexIndent::Document;` in line 27 in the latexindent.pl file, that it can't find it.

Did I make something wrong, or did I miss something?

Full error message:

``Can't locate LatexIndent/Document.pm in @INC (you may need to install the LatexIndent::Document module) (@INC contains: /home/user/opt/anaconda3/envs/test-env/bin /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.32.1 /usr/local/share/perl/5.32.1 /usr/lib/x86_64-linux-gnu/perl5/5.32 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl-base /usr/lib/x86_64-linux-gnu/perl/5.32 /usr/share/perl/5.32 /usr/local/lib/site_perl) at /home/user/opt/anaconda3/envs/test-env/bin/latexindent.pl line 27.
BEGIN failed--compilation aborted at /home/user/opt/anaconda3/envs/test-env/bin/latexindent.pl line 27.
``

All installed packages in the conda env:

```_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
blas                      1.0                         mkl  
brotli                    1.0.9                he6710b0_2  
ca-certificates           2021.10.8            ha878542_0    conda-forge
certifi                   2021.10.8        py39hf3d152e_2    conda-forge
cycler                    0.11.0             pyhd3eb1b0_0  
dbus                      1.13.18              hb2f20db_0  
expat                     2.4.4                h295c915_0  
fontconfig                2.13.1               h6c09931_0  
fonttools                 4.25.0             pyhd3eb1b0_0  
freetype                  2.11.0               h70c0345_0  
giflib                    5.2.1                h7b6447c_0  
glib                      2.69.1               h4ff587b_1  
gst-plugins-base          1.14.0               h8213a91_2  
gstreamer                 1.14.0               h28cd5cc_2  
icu                       58.2                 he6710b0_3  
intel-openmp              2021.4.0          h06a4308_3561  
jpeg                      9e                   h7f8727e_0  
kiwisolver                1.3.2            py39h295c915_0  
latexindent.pl            3.17.2          pl5321hd8ed1ab_0    conda-forge
lcms2                     2.12                 h3be6417_0  
ld_impl_linux-64          2.35.1               h7274673_9  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h5101ec6_17  
libgomp                   9.3.0               h5101ec6_17  
libpng                    1.6.37               hbc83047_0  
libstdcxx-ng              9.3.0               hd4cf53a_17  
libtiff                   4.2.0                h85742a9_0  
libuuid                   1.0.3                h7f8727e_2  
libwebp                   1.2.2                h55f646e_0  
libwebp-base              1.2.2                h7f8727e_0  
libxcb                    1.14                 h7b6447c_0  
libxml2                   2.9.12               h74e7548_1  
lz4-c                     1.9.3                h295c915_1  
matplotlib                3.5.1            py39h06a4308_1  
matplotlib-base           3.5.1            py39ha18d171_1  
mkl                       2021.4.0           h06a4308_640  
mkl-service               2.4.0            py39h7f8727e_0  
mkl_fft                   1.3.1            py39hd3c417c_0  
mkl_random                1.2.2            py39h51133e4_0  
munkres                   1.1.4                      py_0  
ncurses                   6.3                  h7f8727e_2  
numpy                     1.21.5           py39he7a7128_2  
numpy-base                1.21.5           py39hf524024_2  
openssl                   1.1.1o               h7f8727e_0  
packaging                 21.3               pyhd3eb1b0_0  
pcre                      8.45                 h295c915_0  
perl                      5.32.1          0_h7f98852_perl5    conda-forge
perl-carp                 1.50            pl5321hd8ed1ab_0    conda-forge
perl-constant             1.33            pl5321hd8ed1ab_0    conda-forge
perl-exporter             5.74            pl5321hd8ed1ab_0    conda-forge
perl-extutils-makemaker   7.64            pl5321hd8ed1ab_0    conda-forge
perl-file-homedir         1.004           pl5321hd8ed1ab_0    conda-forge
perl-file-path            2.18            pl5321hd8ed1ab_0    conda-forge
perl-file-temp            0.2304          pl5321hd8ed1ab_0    conda-forge
perl-file-which           1.24            pl5321hd8ed1ab_0    conda-forge
perl-parent               0.238           pl5321hd8ed1ab_0    conda-forge
perl-yaml-tiny            1.73            pl5321hd8ed1ab_0    conda-forge
pillow                    9.0.1            py39h22f2fdc_0  
pip                       21.2.4           py39h06a4308_0  
pyparsing                 3.0.4              pyhd3eb1b0_0  
pyqt                      5.9.2            py39h2531618_6  
python                    3.9.12               h12debd9_0  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python_abi                3.9                      2_cp39    conda-forge
qt                        5.9.7                h5867ecd_1  
readline                  8.1.2                h7f8727e_1  
setuptools                61.2.0           py39h06a4308_0  
sip                       4.19.13          py39h295c915_0  
six                       1.16.0             pyhd3eb1b0_1  
sqlite                    3.38.3               hc218d9a_0  
tk                        8.6.11               h1ccaba5_1  
tornado                   6.1              py39h27cfd23_0  
tzdata                    2022a                hda174b7_0  
wheel                     0.37.1             pyhd3eb1b0_0  
xz                        5.2.5                h7f8727e_1  
zlib                      1.2.12               h7f8727e_2  
zstd                      1.4.9                haebb681_0  
```

**Comment by cmhughes** [comment]
Thanks for this.

Everything I know about conda is in https://latexindentpl.readthedocs.io/en/latest/sec-appendices.html#using-conda

**Comment by cmhughes** [comment]
@musm, @tdegeus any ideas?

**Comment by tdegeus** [comment]
@castn Thanks for the report. In the CI and on my system it seems to work. To debug, could you print the output of
```
which latexindent.pl
```
?
Just to clarify, there might be a `latexindent` shipped with texlive, while `latexindent.pl` is shipped with conda. @cmhughes Should we consider unifying this?

**Comment by castn** [comment]
Yeah, `latexindent` is located in my texlive directory, while `latexindent.pl` is located in the conda enviroment.

I tried to use it on a different machine with nearly the same setup, and there it works. 
The difference between the machines is that the one on which latexindent works is not using a custom installation of texlive, it still uses the packages from the debian repositorys. A `conda list` shows me that the only difference on the conda enviroments are that the working one is using `version 3.38.2 of the sqlite conda package` instead of 3.38.3, but my guess is that this might not be the reason that it's not working properly. 

I am quite new to tex and conda so that's what I found out, that might help…

**Comment by castn** [comment]
> Thanks for this.
> 
> Everything I know about conda is in https://latexindentpl.readthedocs.io/en/latest/sec-appendices.html#using-conda

As described in 12.5-12.5.1 in the doc, that's how I installed it.

**Comment by castn** [comment]
> Did you do anything to modify search paths of perl?

Not quite sure what you mean by that. But I don't think that I have changed anything with the path. So it should be still vanilla.

**Comment by castn** [comment]
> Is the correct conda environment fully activated

Yes, it is. Otherwise, I wouldn't find the plugin at all.

**Comment by tdegeus** [comment]
OK. It's just that your @INC really does not look what one would expect in an activated conda environment. I have
```
perl -e "print qq(@INC)"
~/miniforge3/envs/myenv/lib/perl5/5.32/site_perl ~/miniforge3/envs/myenv/lib/perl5/site_perl ~/miniforge3/envs/myenv/lib/perl5/5.32/vendor_perl ~/miniforge3/envs/myenv/lib/perl5/vendor_perl ~/miniforge3/envs/myenv/lib/perl5/5.32/core_perl ~/miniforge3/envs/myenv/lib/perl5/core_perl
```
which indeed locates the location where the module is found:
```
~/miniforge3/envs/myenv/lib/perl5/vendor_perl/LatexIndent
```

**Comment by castn** [comment]
> 

That was the issue! `perl -e "print qq(@INC)"` returned my perl installation of the system, even when I had activated the conda environment. I guess the issue came from the base env, that wasn't really activated by opening a terminal. 

They even warn about that in the conda doc https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#activating-an-environment 
After I deactivated the base environment and activated my custom environment manually, it works fine.

I had to set `conda config --set auto_activate_base False` to deactivate the base environment by default when opening a terminal.

Thanks a lot for the help and the quick response.
